python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -691,7 +691,8 @@ def _flush(self, objs):
 
         for obj in objs:
             # check if the object is dirty, if not do nothing
-            if not obj._dirty:
+            this_id = self.find(obj)
+            if not obj._dirty or this_id not in self._loaded_ids:
```

Comment:
I at least once observed that an EmptyObject had been marked dirty which is a bad thing obviously.
I don't know if this goes away with fixes elsewhere but I want to be belt and braces against loading potentially a large amount of very old jobs into memory on shutdown.